src/github.com/getlantern/flashlight/client/handler.go

```diff
@@ -115,16 +121,11 @@ func (client *Client) intercept(resp http.ResponseWriter, req *http.Request, op
 // pipeData pipes data between the client and proxy connections.  It's also
 // responsible for responding to the initial CONNECT request with a 200 OK.
 func pipeData(clientConn net.Conn, connOut net.Conn, op *ops.Op, closeFunc func()) {
-	writeErrCh := make(chan error, 1)
-	// Start piping from client to proxy
-	op.Go(func() {
-		_, writeErr := io.Copy(connOut, clientConn)
-		writeErrCh <- writeErr
-	})
-
-	// Then start copying from proxy to client.
-	_, readErr := io.Copy(clientConn, connOut)
-	writeErr := <-writeErrCh
```

Comment:
There is no leak.  It's just that we don't close the connections until both the reading and the writing are done.  That meant that we were relying on the idle timeout to close the connections, which would stop the reader and writer.